internal/provider/resources/automation_resource.go

```diff
@@ -0,0 +1,686 @@
+
+	automationID, err := uuid.Parse(plan.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ParseUUIDErrorDiagnostic("Automation", err))
+
+		return
+	}
+
+	err = automationClient.Update(ctx, automationID, updateAutomationRequest)
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Automation", "update", err))
+
+		return
+	}
+
+	updatedAutomation, err := automationClient.Get(ctx, automationID)
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Automation", "get", err))
+
+		return
+	}
+
+	resp.Diagnostics.Append(mapAutomationAPIToTerraform(ctx, updatedAutomation, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *AutomationResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state AutomationResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	automationClient, err := r.client.Automations(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Automation", err))
+
+		return
+	}
+
+	automationID, err := uuid.Parse(state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ParseUUIDErrorDiagnostic("Automation", err))
+
+		return
+	}
+
+	err = automationClient.Delete(ctx, automationID)
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Automation", "delete", err))
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// ImportState imports the resource into Terraform state.
+// Valid import IDs:
+// <automation_id>
+// <automation_id>,<workspace_id>.
+func (r *AutomationResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ",")
+
+	if len(parts) > 2 || len(parts) == 0 {
+		resp.Diagnostics.AddError(
+			"Error importing Automation",
+			"Import ID must be in the format of <automation_id> OR <automation_id>,<workspace_id>",
+		)
+
+		return
+	}
+
+	automationID := parts[0]
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), automationID)...)
+
+	if len(parts) == 2 && parts[1] != "" {
+		workspaceID, err := uuid.Parse(parts[1])
+		if err != nil {
+			resp.Diagnostics.Append(helpers.ParseUUIDErrorDiagnostic("Workspace", err))
+
+			return
+		}
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID.String())...)
+	}
+}
+
+func (r *AutomationResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
```

Comment:
this is a great utility, i'll have to start using it

Reply:
nice! yeah it's a weird, different way of doing validation - in this case, I wanted only one of these keys to exist, but couldn't find a way to do it in the Schema directly